### ISSUE TITLE
hawck-add: "install: cannot stat '/tmp/...': Permission denied

### ISSUE THREAD

**Issue by eddy-geek** [snyball/hawck#74]
**To Reproduce**
1. Follow install steps
2.
```
> hawck-add ~/.config/hawck/scripts/example.hwk
New keys added, require authentication for whitelisting them ...
[sudo] password for $USER: 
install: cannot stat '/tmp/$USER/tmp.rJVo1HvUPe': Permission denied
```
3. Script does nor work

**Expected behavior**
no error and script works

**Desktop (please complete the following information):**
- Distro: Ubuntu 22.04 / kernel 5.19.0-24-generic
- Version: git today (rev `8145084`)
- Method of installation: `install.sh`
- Keymap: custom
- Keyboard : Laptop built-in

**Additional context**

```
> cat /usr/bin/hawck-add
HAWCKD_INPUT_USER=hawck-input-share
...
command sudo --user="$HAWCKD_INPUT_USER" -- install -m 644 "$tmp_keys" "$real_keys"

> groups
$USER ... hawck-input-share

> groups hawck-input
hawck-input : hawck-input input hawck-input-share uinput

> ll '/tmp/$USER/tmp.YL3bg7XOsv'
Permissions Size User       Date Modified  Name
.rw-r--r--   126 $USER  4 janv. 11:38 /tmp/$USER/tmp.YL3bg7XOsv
```

I can reproduce with
```
> sudo --user=hawck-input -- install -m 644 '/tmp/$USER/tmp.YL3bg7XOsv' /var/lib/hawck-input/keys/example.csv
install: cannot stat '/tmp/$USER/tmp.YL3bg7XOsv': Permission denied
```

**Comment by snyball** [comment]
Are you sure only your user got added to the group? Check with `groups hawck-input`.

If not, then then something might have gone wrong with the `post_install` hook, try this:

```bash
$ sudo -E bash # must be root, and must be bash or zsh shell (not fish/csh/ksh/etcsh)
$ source bin/hawck-git/hawck-git.install # make sure to clone with --recurse-submodules, otherwise you won't have the hawck-git.install file
$ post_install

**Comment by eddy-geek** [comment]
I think this is the issue, tmp folder is private
```
ls -ld '/tmp/$USER'
drwx------ 2 $USER $USER 4096 janv.  4 11:41 /tmp/$USER
```

**Comment by snyball** [comment]
Interesting, is the `/tmp/` directory itself private? I don't recall `hawck-add` needing to save anything to `/tmp/$(whoami)`, but maybe `mktemp` has a different behaviour on Ubuntu?

```fish
jonas@lambda ~> ls -al /tmp/
total 416
drwxrwxrwt 18 root  root     440 Jan  4 11:48 ./
...
```

**Comment by eddy-geek** [comment]
Indeed, `chmod go+x /tmp/$USER` fixes it
this may have been a security hardening by ubuntu

**Comment by snyball** [comment]
@eddy-geek Just saw your reply now. It seems like Ubuntu must've patched `mktemp` to save to `/tmp/$(whoami)/` (rather than just using the existing `$XDG_RUNTIME_DIR`, for some reason?) Then you're running into a permissions issue because `command sudo --user="hawck-inputd" -- install -m 644 "$tmp_keys" "$real_keys"` can't access that directory, because _that_ command is running under the `hawck-input` user.

In this case I think it makes sense for the hawck setup scripts to create a `/var/lib/hawck-inputd/tmp` which is marked `rwxrwx---` with group `hawck-input-share`, and to place the temporary file there.

**Comment by snyball** [comment]
Should be fixed by 067fce1276849b1d287019e4ea38a671ae31b9b0
